js/ui/iconGrid.js

```diff
@@ -157,10 +182,23 @@ const BaseIcon = new Lang.Class({
     _createIconTexture: function(size) {
         if (this.icon)
             this.icon.destroy();
+        this.extraIcons.forEach(function (i) {i.destroy()});
```

Comment:
Missing semicolon after `destroy()`, but please also add a space before and after the curly braces.